modules/services/autossh.nix

```diff
@@ -91,7 +92,7 @@ in
               # How often AutoSSH checks the network, in seconds
               environment.AUTOSSH_POLL="30";
 
-              command = "${pkgs.autossh}/bin/autossh -M ${toString mport} ${s.extraArguments}";
+              command = [ (lib.getExe pkgs.autossh) "-M" "${toString mport}" ] ++ s.extraArguments;
```

Comment:
Looks like this is the only place where we pass arbitrary user arguments in this field. After this change users will have to adjust their configurations and wouldn’t be able to reference environment variables here any more. I’m okay with it if you are, but I’m guessing we might have services for which this is undesirable in future.

Reply:
In those situations, I think we should direct them to set `serviceConfig.ProgramArguments` directly (possibly with `lib.mkAfter`)

I did a quick GitHub Code search and didn't see any configs that used environment variables for `services.autossh.extraArguments`:

https://github.com/search?utf8=✓&q=path%3A*.nix+services+autossh+extraArguments&type=Code